hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirectory.java

```diff
@@ -1860,6 +1928,21 @@ FSPermissionChecker getPermissionChecker()
     }
   }
 
+  public INode getInode(INode inode) {
+    return inodeMap.get(inode);
+  }
+  public INode getInodeFromTempINodeMap(long id) {
+    LOG.debug("getInodeFromTempINodeMap: id={}, TempINodeMap.size={}",
+        id, inodeMapTemp.size());
+    /*
+     * Convert a long inode id into an INode object. We only need to compare
+     * two inodes by inode id. So, it can be any type of INode object.
+     */
+    INode inode = new INodeDirectory(id, null,
```

Comment:
This will be a lot of new object instantiations during loading.
Can we just define tempInodeMap : id -> INode
to avoid creating new objects

Reply:
GSet<K, V> requires V to be extended from K. Thus, we replaced GSet<Inode, InodeWithAdditionalFields> with HashMap<Long, InodeWithAdditionalFields>.